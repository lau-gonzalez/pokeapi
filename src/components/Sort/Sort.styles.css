.sort-container {
  display: flex;
  gap: 20px;
}

.selector-container {
  display: flex;
  width: 200px;
}
