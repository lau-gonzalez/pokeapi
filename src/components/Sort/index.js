import React, { useEffect, useState } from "react";
import "./Sort.styles.css";

const Sort = ({ sortCards }) => {
  const [sortBy, setSortBy] = useState("id");
  const [order, setOrder] = useState("ascendant");

  const handleSortByChange = (e) => setSortBy(e.target.value);

  const handleOrderChange = (e) => setOrder(e.target.value);

  useEffect(() => {
    sortCards(sortBy, order);
  }, [sortBy, order]); //eslint-disable-line

  return (
    <div className="sort-container">
      <div className="form-floating">
        <select
          className="form-select"
          id="sortBySelect"
          onChange={handleSortByChange}
        >
          <option value="id">Number</option>
          <option value="name">Name</option>
        </select>
        <label>Sort By</label>
      </div>

      <div className="form-floating">
        <select
          className="form-select"
          id="orderSelect"
          onChange={handleOrderChange}
        >
          <option value="ascendant">Ascendant</option>
          <option value="descendant">Descendant</option>
        </select>
        <label>Order</label>
      </div>
    </div>
  );
};

export default Sort;
