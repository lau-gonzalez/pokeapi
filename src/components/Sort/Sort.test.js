import { render, cleanup, screen } from "@testing-library/react";
import Sort from "./index";

afterEach(cleanup);

const sortCards = jest.fn();

describe("<Sort/>", () => {
  it("renders the component", () => {
    const { asFragment } = render(<Sort sortCards={sortCards} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the sort by selector", () => {
    render(<Sort sortCards={sortCards} />);
    expect(screen.getByText("Number")).toBeInTheDocument();
    expect(screen.getByText("Sort By")).toBeInTheDocument();
  });

  it("should render the order selector", () => {
    render(<Sort sortCards={sortCards} />);
    expect(screen.getByText("Ascendant")).toBeInTheDocument();
    expect(screen.getByText("Order")).toBeInTheDocument();
  });
});
