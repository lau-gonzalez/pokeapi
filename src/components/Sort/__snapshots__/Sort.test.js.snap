// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sort/> renders the component 1`] = `
<DocumentFragment>
  <div
    class="sort-container"
  >
    <div
      class="form-floating"
    >
      <select
        class="form-select"
        id="sortBySelect"
      >
        <option
          value="id"
        >
          Number
        </option>
        <option
          value="name"
        >
          Name
        </option>
      </select>
      <label>
        Sort By
      </label>
    </div>
    <div
      class="form-floating"
    >
      <select
        class="form-select"
        id="orderSelect"
      >
        <option
          value="ascendant"
        >
          Ascendant
        </option>
        <option
          value="descendant"
        >
          Descendant
        </option>
      </select>
      <label>
        Order
      </label>
    </div>
  </div>
</DocumentFragment>
`;
